src/ray/core_worker/actor_manager.h

```diff
@@ -196,6 +200,9 @@ class ActorManager {
   /// getting them from GCS frequently.
   absl::flat_hash_map<std::string, ActorID> cached_actor_name_to_ids_
       GUARDED_BY(cache_mutex_);
+
+  /// Mutex of actor's state subscription.
+  mutable absl::Mutex subscribe_mutex_;
```

Comment:
Not sure why we need this if the flag is also atomic.

Reply:
> Not sure why we need this if the flag is also atomic.

This mutext is used to protect that the subscription only invoked once.
As @kfstorm suggested, I use a `absl::flat_hash_set<ActorID>` to record that if the actor state is subscribed, and the mutext will be used to protect `absl::flat_hash_set<ActorID>`.